matlab_simulation/08-planning/example_search_graph.m

```diff
@@ -0,0 +1,47 @@
+% Get same problem every time (fix random number generator start point)
+rng('default')
+
+%% Generate a graph
+n = 30; % Nodes
+s = 100; % Space
+% Random nodes
+nodes = s*rand(n,2);
+e = sparse(zeros(n,n));
+D = sparse(zeros*ones(n,n));
+
+% Add closest p edges
+p = 5;
+for i = 1:n
+    for j = 1:n
+        d(j) = norm(nodes(i,:)-nodes(j,:));
+    end
+    [d2,ind] = sort(d);
+    for j=1:p
+        if (i~=ind(j))
+            e(i,ind(j)) = 1;
+            e(ind(j),i) = 1;
+            D(i,ind(j)) = d(ind(j));
+            D(ind(j),i) = d(ind(j));
+        end
+    end
+end
+
+% Pick furthest apart start and end node
+dmax = 0;
+start = 0; finish = 0;
+for i=1:n

```

Comment:
Could the loop to 'Pick furthest apart start and end nodes' be fused with the 'add closest p edges' loop so that we dont have to run a new loop at [line 32]. It seems like the norm is being calculated again and there might not be a need for it.